Added screenshot hotkey

sardap opened the pull request:
Added screenshot hotkey and added a toggle to debug GUI to disable screenshot hotkey #1765

ManDude commented on game/graphics/pipelines/opengl.cpp:
```diff
@@ -435,7 +460,9 @@ void render_game_frame(int game_width,
+    if (g_gfx_data->debug_gui.get_screenshot_flag() || want_hotkey_screenshot) {
       options.save_screenshot = true;
       options.game_res_w = g_gfx_data->debug_gui.screenshot_width;
       options.game_res_h = g_gfx_data->debug_gui.screenshot_height;
```
Don't use this code for quick screenshots, as they need to be set using imgui which is exactly what we want to avoid.

sardap replied:
Okay I have separated them

ManDude commented on game/graphics/pipelines/opengl.cpp:
```diff
@@ -389,12 +402,24 @@ void GLDisplay::on_iconify(GLFWwindow* /*window*/, int iconified) {
-std::string make_output_file_name(const std::string& file_name) {
+std::string make_full_screenshot_output_file_path(const std::string& file_name) {
   file_util::create_dir_if_needed(file_util::get_file_path({"gfx_dumps"}));
   return file_util::get_file_path({"gfx_dumps", file_name});
```
we can probably rename this folder to `screenshots` now.

sardap replied:
Done

sardap commented:
@ManDude Do I resolve the conversation after addressing the comment? Or do I leave that to the person who left the comment? 

I have only used bitbucket for pull requests and it doesn't have a concept of resolving a conversation.

xTVaser commented:
Doesn't really matter, resolving the comment just collapses it to minimize vertical space for everyone.  It can be unresolved again if deemed that it wasn't actually resolved.

ManDude commented on game/graphics/pipelines/opengl.cpp:
```diff
@@ -389,12 +402,24 @@ void GLDisplay::on_iconify(GLFWwindow* /*window*/, int iconified) {
 }
 }  // namespace
 
+static std::string get_current_timestamp() {
+  auto current_time = std::time(nullptr);
+  auto local_current_time = *std::localtime(&current_time);
+  std::ostringstream oss;
+  oss << std::put_time(&local_current_time, "%Y_%m_%d_%H_%M_%S");
+  return oss.str();
+}
```
Looking at `sceCdReadClock`, we do the following to get the time:

```cpp
  time_t t = time(0);
  std::tm* date = localtime(&t);
```

Stringstreams are kinda janky and we haven't used them anywhere, so it might be better to just format the string using that code.

sardap replied:
I opted to use strftime instead.  fmt has date formatting support but we are missing "fmt/chrono.h" https://fmt.dev/latest/api.html

sardap commented:
Fixed clang formatting error sorry for missing it have set up a pre hook to catch it next time

water111 reviewed (approved):
Thanks!
